datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -2008,6 +2009,59 @@ public void validateDatastreamsUpdate(List<Datastream> datastreams) throws Datas
     }
   }
 
+  /**
+   * Verify certain checks before proceeding with datastream update. Compares few fields with existing datastream to
+   * check if update can be allowed or not
+   * @param datastream New state of datastream
+   * @param oldDatastream Current datastream
+   * @throws DatastreamValidationException if update cannot be allowed
+   */
+  public void verifyUpdateDatastreamChecks(Datastream datastream, Datastream oldDatastream) throws DatastreamValidationException {
+    String key = datastream.getName();
+    if (!oldDatastream.hasConnectorName() || !datastream.hasConnectorName()) {
+      throw new DatastreamValidationException(String.format("Failed to update %s because connector is not present."
+          + " Are they valid? old: %s, new: %s", key, oldDatastream, datastream));
+    }
+    if (!datastream.getConnectorName().equals(oldDatastream.getConnectorName())) {
+      throw new DatastreamValidationException(String.format("Failed to update %s. Can't update connector in update request."
+          + " old: %s, new: %s", key, oldDatastream, datastream));
+    }
+    if (!oldDatastream.hasTransportProviderName() || !datastream.hasTransportProviderName()) {
+      throw new DatastreamValidationException(String.format("Failed to update %s. Can't update transport provider in"
+          + " update request. old: %s, new: %s", key, oldDatastream, datastream));
+    }
+    if (!datastream.getTransportProviderName().equals(oldDatastream.getTransportProviderName())) {
+      throw new DatastreamValidationException(String.format("Failed to update %s. Can't update transport provider in"
+          + " update request. old: %s new: %s", key, oldDatastream, datastream));
+    }
+    if (!oldDatastream.hasDestination() || !datastream.hasDestination()) {
+      throw new DatastreamValidationException(String.format("Failed to update %s because destination is not set. "
+          + "Are they initialized? old: %s, new: %s", key, oldDatastream, datastream));
+    }
+
+    if (!oldDatastream.hasStatus() || !datastream.hasStatus()) {
+      throw new DatastreamValidationException(String.format("Failed to update %s because status is not present."
+          + " Are they valid? old: %s, new: %s", key, oldDatastream, datastream));
+    }
+    if (!datastream.getStatus().equals(oldDatastream.getStatus())) {
+      throw new DatastreamValidationException(String.format("Failed to update %s. Can't update status in update request."
+          + " old: %s new: %s", key, oldDatastream, datastream));
+    }
+
+    if (datastream.getMetadata().containsKey(NUM_TASKS) &&
+        !datastream.getMetadata().get(NUM_TASKS).equals(oldDatastream.getMetadata().get(NUM_TASKS))) {
+      throw new DatastreamValidationException(String.format("Failed to update %s. Can't update numTasks."
+          + " old: %s new: %s", key, oldDatastream, datastream));
+    }
+
+    if (!_config.getOverrideDatastreamUpdateChecks()) {
```

Comment:
So this flag only allows you to bypass the destination check? Why is this needed and if it is should we rename it?

Reply:
Yes for now we only bypass destination change check. For IPv6, we plan to manually (through brooklin tool command) update destination of the exisiting datastreams, and this check prevents that. 
This is only a temporary change. Once IPv6 migration is done we will revert this change. So not a lot of benefit in renaming it if you dont feel too strongly about it

Reply:
If the plan is to revert that's fine